Added the Nod repair bay icon

reaperrr opened the pull request:
Set up Nod repair bay icon for cases where Nod captures a GDI construction yard.

reaperrr committed:
Use Nod variant of repair bay icon

Mailaender commented on mods/ts/rules/palettes.yaml:
```diff
@@ -22,7 +22,6 @@
 		Name: chrome
 		Filename: cameo.pal
 		AllowModifiers: false
-		ShadowIndex: 242
```
Sadly this breaks the clock:

![image](https://cloud.githubusercontent.com/assets/756669/9040116/355bf4a4-3a00-11e5-99a1-ff5eaab50762.png)

GraionDilach replied:
The proper fix of this would be to decouple the clock palette from the underlaying icon. I had started to work on this, but my last attempt broke support powers. >_>

pchote replied:
The easiest fix would be to specify the clock palette on the palette / support power widgets themselves.  There's no need for a per-icon clock sequence or clock sequence palette.

reaperrr replied:
Damn...

Anyway, the problem here is that Westwood did not really use regular shadow, but rather translucency, and iirc TS clock uses `anim.pal` rather than `cameo.pal`, too. So I guess the correct solution will be to add a `ClockPalette` to `Buildable`.

GraionDilach replied:
Indeed that was my plan as well - along with per-widget clock sequences - the whole thing was a PEBKAC.

pchote replied:
As I mentioned above, put it on the widgets instead of `Buildable` please.

pchote commented:
Do we have a GDI icon for the Nod repair bay?  If not, then I think we should keep it as-is for symmetry and to indicate which buildings are captured tech (or go through and remap all of the rest ourselves).

reaperrr commented:
> Do we have a GDI icon for the Nod repair bay?

There is no Nod repair bay, Nod only has a mobile repair vehicle.

GraionDilach commented:
:+1: on the Nod repair bay though.

Mailaender commented:
> Nod players could capture a GDI construction yard, so let's use the Nod version of its icon.

If you follow that logic, you also need Nod versions for the tech center, turrets etc.

reaperrr commented:
> If you follow that logic, you also need Nod versions for the tech center, turrets etc.

I meant that in the way of "the nod icon is there and we already use all other faction-specific icons, so let's use this one, too".

Mailaender commented:
Well, but to use it, you need to duplicate it with a fixed up version so I am not sure that counts. https://github.com/reaperrr/OpenRA/blob/ts-icons1/mods/ts/bits/nfixicon.shp ;)

reaperrr commented:
Using a custom sequence offset would work, too. I just felt this was a cleaner solution.

reaperrr commented:
Updated.

- Decided to remove the custom icon shp and used offsets to correct position ingame.
- Added UI unhardcoding of icon clock and darkening.

Mailaender commented on OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs:
```diff
@@ -46,6 +48,13 @@ public enum ReadyTextStyleOptions { Solid, AlternatingColor, Blinking }
+		public readonly string ClockPalette = ChromeMetrics.Get<string>("ProductionClockPalette");
+		public readonly string NotBuildableAnimation = ChromeMetrics.Get<string>("ProductionNotBuildableAnimation");
+		public readonly string NotBuildableSequence = ChromeMetrics.Get<string>("ProductionNotBuildableSequence");
+		public readonly string NotBuildablePalette = ChromeMetrics.Get<string>("ProductionNotBuildablePalette");
```
Why use chrome metrics here? Just put it on the widget and leave the defaults to work with RA/TD.

GraionDilach replied:
I'd say these could be moved to `ProductionQueue`/`SupportPowerManager` instead.

Mailaender replied:
No, they are UI related just like ReadyText and HoldText.

reaperrr replied:
@pchote said on irc I should use ChromeMetrics...

Mailaender replied:
We only dropped that there for generic re-usable parts of the UI that is why I was asking. Also this will break every custom mods as you don't have upgrade rules to add the missing bits.

reaperrr commented:
Removed everything except the Nod repair bay icon changes, the rest will come back later in a separate PR when I have more time.

reaperrr commented:
Counted Graion's +1 for the icon.

Mailaender commented:
I submitted https://github.com/OpenRA/OpenRA/pull/9129 with my suggested changes.
